src/facilities/models.py

```diff
@@ -0,0 +1,171 @@
+    """
+
+    class Meta:
+        unique_together = [("slug", "responsible_team")]
+
+    name = models.CharField(max_length=100, help_text="The name of this facility type")
+
+    slug = models.SlugField(
+        blank=True,
+        help_text="The url slug for this facility type. Leave blank to autogenerate one.",
+    )
+
+    description = models.TextField(help_text="Description of this facility type")
+
+    icon = models.CharField(
+        max_length=100,
+        default="fas fa-list",
+        blank=True,
+        help_text="Name of the fontawesome icon to use, including the 'fab fa-' or 'fas fa-' part.",
+    )
+
+    responsible_team = models.ForeignKey(
+        "teams.Team",
+        on_delete=models.PROTECT,
+        help_text="The Team responsible for this type of facility. This team will get the notification when we get a new FacilityFeedback for a Facility of this type.",
+    )
+
+    quickfeedback_options = models.ManyToManyField(
+        to="facilities.FacilityQuickFeedback",
+        help_text="Pick the quick feedback options the user should be presented with when submitting Feedback for a Facility of this type. Pick at least the 'N/A' option if none of the other applies.",
+    )
+
+    @property
+    def camp(self):
+        return self.responsible_team.camp
+
+    camp_filter = "responsible_team__camp"
+
+    def __str__(self):
+        return f"{self.name} ({self.camp})"
+
+    def save(self, **kwargs):
+        if not self.slug:
+            self.slug = slugify(self.name)
+        if not self.slug:
+            raise ValidationError("Unable to slugify")
+        super().save(**kwargs)
+
+
+class Facility(CampRelatedModel, UUIDModel):
+    """
+    Facilities are toilets, thrashcans, cooking and dishwashing areas, and any other part of the event which could need attention or maintenance.
+    """
+
+    facility_type = models.ForeignKey(
+        "facilities.FacilityType", related_name="facilities", on_delete=models.PROTECT
+    )
+
+    name = models.CharField(
+        max_length=100, help_text="Name or description of this facility",
+    )
+
+    description = models.TextField(help_text="Description of this facility")
+
+    @property
+    def team(self):
+        return self.facility_type.responsible_team
+
+    @property
+    def camp(self):
+        return self.facility_type.camp
+
+    camp_filter = "facility_type__responsible_team__camp"
+
+    def __str__(self):
+        return self.name
+
+
+class FacilityFeedback(CampRelatedModel):
+    """
+    This model contains participant feedback for Facilities.
+    It is linked to the user and the facility, and to the
+    quick_feedback choice the user picked (if any).
+    """
+
+    user = models.ForeignKey(
+        "auth.User",
+        null=True,
+        blank=True,
+        on_delete=models.PROTECT,
+        related_name="facility_feebacks",
+        help_text="The User this feedback came from, empty if the user submits anonymously",
+    )
```

Comment:
Can this be a ManyToMany field or can there be an additional field to collect all the +1s? It could be used to suggest urgency of an issue when it's reported by many people..

Reply:
Currently the feedback is not readable to participants or people outside of the team responsible for that facility type. I was not sure about this. We want to be open, but we also want to maintain some level of control over what we display on the website. If the facilityfeedback is going to be visible to participants we need some way of approving it.

Reply:
Ah yes, maybe the aggregation can happen somewhere else then. I suppose that each feedback will then be unique per user. This could, though, generate a lot of data.

But as long as it exists, that data can be aggregated in a single migration to a new model later on, if need be.